satpy/tests/utils.py

```diff
@@ -355,3 +355,100 @@ def __call__(self, dsk, keys, **kwargs):
             raise RuntimeError("Too many dask computations were scheduled: "
                                "{}".format(self.total_computes))
         return dask.get(dsk, keys, **kwargs)
+
+
+def make_a_scene(cont_dict, daskify=False, area=True, common_attrs=None):
+    """Make a fake Scene.
+
+    Make a Scene from fake data.  Data are provided in the
+    ``cont_dict`` argument.  In ``cont_dict``, keys should be strings
+    or DatasetID/DataID, and values may be either numpy.ndarray or
+    xarray.DataArray, in either case with exactly two dimensions.
+    The function will convert each of the numpy.ndarray objects into
+    an xarray.DataArray and assign those as datasets to a Scene object.
+    A fake AreaDefinition will be assigned for each array, unless disabled
+    by passing ``area=False``.  When areas are automatically generated,
+    arrays with the same shape will get the same area.
+
+    This function is exclusively intended for testing purposes.
+
+    If regular ndarrays are passed and the keyword argument daskify is
+    True, DataArrays will be created as dask arrays.  If False (default),
+    regular DataArrays will be created.  When the user passes xarray.DataArray
+    objects then this flag has no effect.
+
+    Args:
+        cont_dict: mapping with str/datasetid/dataid to ndarrays
+        daskify, optional: boolean, use dask or not, default False
+        area, optional: boolean, add areas, default True
```

Comment:
Could you make this accept ready instances of areadef or swathdefs too ?

Reply:
Now accepts any instance of `BaseDefinition`, which will be assigned directly to the `area` attribute.